### PR TITLE
[Bugfix:Autograding] Tolerance parse fix

### PR DESCRIPTION
**Pull request by micpap25**
Makes a fix on #8251 so it works with numbers that have non-period punctuation surrounding them will still be recognized as numbers.

### CODE REVIEW

**Review by dorian451 — APPROVED** [review]
Looks good to me

**Review by ziesski — APPROVED** [review]
Reviewed the code, and looks like it will recognize numbers numbers surrounded by non-period punctation, as described. Since it is cpp code, Barb should take another look before merging.

File: grading/clean.cpp [hunk 1]
```diff
@@ -308,7 +313,23 @@ bool isNumber(const std::string &str) {
...
+  return atLeastOneDigit;
+}
+
+std::string isolateAlphanumAndNumberPunctuation(const std::string &str) {
+  std::string::const_iterator begin = str.begin();
+  while (!isalnum(*begin) && !(*begin == '.')) {
+    begin++;
+    if (begin == str.end())
+    {
+      return str;
```
Comment (bmcutler) [inline grading/clean.cpp:325]: so you return the whole string if there is no number at all?
(this is not intuitive)

File: grading/clean.cpp [hunk 2]
```diff
@@ -308,7 +313,23 @@ bool isNumber(const std::string &str) {
       return false;
     }
   }
-  return true == atLeastOneDigit;
+  return atLeastOneDigit;
+}
+
+std::string isolateAlphanumAndNumberPunctuation(const std::string &str) {
```
Comment (bmcutler) [inline grading/clean.cpp:319]: this function is somewhat confusing in purpose and should at least have a comment at the start...

**Review by bmcutler — CHANGES_REQUESTED** [review]

**Review by bmcutler — APPROVED** [review]
thanks!

**Pull request merged**